facebook/delphiFacebook/R/contingency_calculate.R

```diff
@@ -104,3 +104,53 @@ post_convert_count_to_pct <- function(df) {
   return(mutate(df,
                 val = 100 * .data$val / sum(.data$val, na.rm=TRUE)))
 }
+
+#' Return numeric response estimates. Val is the mean of a numeric vector.
+#'
+#' This function takes vectors as input and computes the response values
+#' (a point estimate named "val" and 25, 50, and 75th percentiles).
+#'
+#' @param response a vector of multiple choice responses
+#' @param weight a vector of sample weights for inverse probability weighting;
+#'   invariant up to a scaling factor
+#' @param sample_size The sample size to use, which may be a non-integer (as
+#'   responses from ZIPs that span geographical boundaries are weighted
+#'   proportionately, and survey weights may also be applied)
+#' @param total_represented Number of people represented in sample, which may
+#'   be a non-integer
+#'
+#' @return a list of named mean and other descriptive statistics
+#'
+#' @importFrom survey svydesign svymean svyvar oldsvyquantile
+compute_numeric_mean <- function(response, weight, sample_size, total_represented) {
+  assert(length(response) == length(weight))
+  
+  if (length(response) == 1) {
+    # svydesign complains if given only one response, so return an empty df
+    # instead. This group will be dropped anyway.
+    return(list(
+      val = NA_real_,
+      se = NA_real_,
+      sd = NA_real_,
+      p25 = NA_real_,
+      p50 = NA_real_,
+      p75 = NA_real_,
+      sample_size = NA_real_,
+      effective_sample_size = NA_real_,
+      represented = NA_real_
+    ))
+  }
+  
+  design <- svydesign(id = ~1, weight = ~weight, data = data.frame(response, weight))
+  return(list(
+    val = as.data.frame(svymean(~response, na.rm = TRUE, design = design))[,"mean"],
+    se = NA_real_,
+    sd = as.data.frame(sqrt(svyvar(~response, na.rm = TRUE, design = design)))[,"variance"],
+    p25 = as.data.frame(oldsvyquantile(~response, na.rm = TRUE, design = design, quantiles = 0.25))[,"0.25"],
```

Comment:
Hmm, why are we using `oldsvyquantile` instead of `svyquantile`? I don't know the details of what changed

Reply:
Incidentally you might like [srvyr](http://gdfe.co/srvyr/index.html), which provides a tidy interface to the survey package

Reply:
`svyquantile` doesn't work for older versions of R, so Esther switched her code to use the `old` version of the function. [Context](https://app.asana.com/0/0/1202106541327630/1202179687386809/f). Looks like the old one is slower and less flexible.

The approach I implemented here is just a simplified version of [Esther's code](https://app.asana.com/0/0/1202106541327630/1202143955278131/f), since we already have sample size filters, e.g., elsewhere.